src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs

```diff
@@ -94,9 +98,25 @@ public override async Task<ScriptSecrets> ReadAsync(ScriptSecretsType type, stri
                     }
                 }
             }
+            catch (RequestFailedException rfex) when (rfex.Status == 409)
+            {
+                // If the read operation failed because the blob access tier is set to archived, log a diagnostic event.
+                if (rfex.ErrorCode.Equals(BlobArchivedName, StringComparison.OrdinalIgnoreCase))
+                {
+                    DiagnosticEventLoggerExtensions.LogDiagnosticEventError(
```

Comment:
you can do `logger.LogDiagnosticEventError` here instead.

Reply:
👍 consider adding a `private readonly ILogger` in the constructor to support this and use _logger instead.

Reply:
Fixed. I will use the existing `ILogger` from the base class. 

https://github.com/Azure/azure-functions-host/blob/6ae170c32c97864d33424e1f8cc1783acdbc907a/src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs#L30

which is defined here:
https://github.com/Azure/azure-functions-host/blob/6ae170c32c97864d33424e1f8cc1783acdbc907a/src/WebJobs.Script.WebHost/Security/KeyManagement/BaseSecretsRepository.cs#L43